### PR TITLE
[WA-41]: forecast 3 days

### PR DESCRIPTION
**Pull request by ketryin**

### CODE REVIEW

File: src/app/forecast/forecast.component.ts [hunk 1]
```diff
@@ -3,71 +3,49 @@ import { Component, OnInit } from '@angular/core';
...
+  forecastDays: number;
+  weatherInfo: CityWeatherInfo;
 
-  forecastBreadcrumbLinks: BreadcrumbLink[] = [
-    { link: '/', name: 'Home', isActive: false },
-    { link: '/kiev/details', name: 'Details', isActive: false },
-    { link: '/kiev/details', name: 'Forecast', isActive: true },
-  ];
+  constructor(
+    activateRoute: ActivatedRoute,
```
[the remark this answers is not in the document]
Comment (ketryin) [reply src/app/forecast/forecast.component.ts:22]: I need to initialize internal state of component and constructor seems like a reasonable place to initialize a state. Maybe you can suggest better solution?

File: src/app/details/details.component.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
   class="details__current-weather"
   [weatherInfo]="weatherInfo"
 ></app-current-weather>
-<app-btn-groups class="details__details-btns"></app-btn-groups>
+<app-btn-groups class="details__details-btns" [weatherInfo]="weatherInfo"></app-btn-groups>
```
Comment (dmitry-chirva) [inline src/app/details/details.component.html:6]: we don't need to send data in btn-groups

File: src/app/details/details.component.ts [hunk 3]
```diff
@@ -30,28 +24,35 @@ export class DetailsComponent {
     humidity: [],
     pressure: [],
   };
+  city : string;
```
Comment (dmitry-chirva) [inline src/app/details/details.component.ts:27]: 1) rename to currentCity
2) add please private modificator

File: src/app/forecast/forecast.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitry-chirva) [reply src/app/forecast/forecast.component.ts:22]: add mofificator private

File: src/app/details/details.component.ts [hunk 4]
```diff
@@ -30,28 +24,35 @@ export class DetailsComponent {
...
+  weatherInfo: CityWeatherInfo | null = null;
+  detailBreadcrumbLinks: BreadcrumbLink[] = [];
 
   constructor(
+    activateRoute : ActivatedRoute,
```
Comment (dmitry-chirva) [inline src/app/details/details.component.ts:32]: private

File: src/app/forecast/forecast.component.ts [hunk 5]
```diff
@@ -3,71 +3,49 @@ import { Component, OnInit } from '@angular/core';
...
-      weatherIcon: 'assets/images/weather.png',
-      additionalInfo: {
-        weatherLabel: 'Light snow',
-        windSpeed: '10 km/h',
-        humidity: '90%',
-      },
-    },
-  ];
+  getForecastDays(forecastType : string): number {
+    if (forecastType === 'three-days') {
```
Comment (dmitry-chirva) [inline src/app/forecast/forecast.component.ts:43]: create constant

forecastType = Object.frezee({
 ['three-days']: 3,
 ['ten-days']: 10,
 default: 1
})

then 
getForecastDays(type: string): number {
return forcastType[type] ? forcastType[type] || forcastType.default;

}

File: src/app/shared/components/btn-groups/btn-groups.component.html [hunk 6]
```diff
@@ -1,24 +1,24 @@
 <div class="btn-groups">
   <button
-    routerLink="/kiev/details"
+    routerLink="/{{weatherInfo?.city}}/details"
```
Comment (dmitry-chirva) [inline src/app/shared/components/btn-groups/btn-groups.component.html:3]: ok we don't need refactoring.
just rename btn-groups to forecast-btn-group.
And one more thing here we need use only city

File: src/core/api/common/mapper-module.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { CityWeatherInfo } from 'src/app/shared/interfaces/city-weather-info.interfaces';
```
Comment (dmitry-chirva) [inline src/core/api/common/mapper-module.ts:1]: We have two options create utils class for this or create service.
For example rename mapper-module to weather-transform.service

File: src/app/details/details.component.html (hunk 2 above)
Comment (ketryin) [reply src/app/details/details.component.html:6]: but we need to create links for forecasts, if we won't pass weatherInfo with city, there's no way we can determine for which city we should render forecast

File: src/app/forecast/forecast.component.ts [hunk 8]
```diff
@@ -2,72 +2,56 @@ import { Component, OnInit } from '@angular/core';
...
+  forecastBreadcrumbLinks: BreadcrumbLink[] = [];
+  forecastDays: number;
+  weatherInfo: CityWeatherInfo;
 
-  forecastBreadcrumbLinks: BreadcrumbLink[] = [
-    { link: '/', name: 'Home', isActive: false },
-    { link: '/kiev/details', name: 'Details', isActive: false },
-    { link: '/kiev/details', name: 'Forecast', isActive: true },
-  ];
+  forecastTypes : any = Object.freeze({
```
Comment (dmitry-chirva) [inline src/app/forecast/forecast.component.ts:20]: i mean in other file
forecast-types.constant.ts

File: src/app/shared/components/forecast-btn-group/btn-groups.component.ts [hunk 9]
```diff
@@ -1,9 +1,10 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 
 @Component({
   selector: 'app-btn-groups',
   templateUrl: './btn-groups.component.html',
   styleUrls: ['./btn-groups.component.scss']
 })
 export class BtnGroupsComponent {
+  @Input() currentCity : string | undefined = undefined;
```
Comment (dmitry-chirva) [inline src/app/shared/components/forecast-btn-group/btn-groups.component.ts:9]: minor: i suggest to add default string instead of undefined

**Pull request merged**